### ISSUE TITLE
2 bugs will occur at 100 thousand years later and 2.1 billion years later

### ISSUE THREAD

**Issue by kshadow1000** [xorg62/tty-clock#113]
100 thousand years later the date will take 2 lines.
![IMG_20231222_225554](https://github.com/xorg62/tty-clock/assets/54498130/00870506-993a-40d4-b421-1746012187d9)

2.1 billion years later a "Segmentation fault." will occur.

**Comment by Cheaterman** [comment]
That gives them plenty of time to fix it, then.

**Comment by AmyrAhmady** [comment]
I'm coming from future and I can confirm tty-clock does not exist anymore. I would've stayed here to talk but I'm afraid I must go right now.

**Comment by izmyname** [comment]
I just came from the future to confirm 2.1 billion problem. Please, hoomans, fix it, asap, as people here grow suspicious to giant sentient tarantulas, from the future and it's hard to counter the gravitation and the Sun being small and yellow here makes me scary.

**Comment by livcher** [comment]
Longtermists be like:

**Comment by pompompur1nn** [comment]
yeah. Can confirm this exploded 900 systems. In the future. Please fix now.

**Comment by Jelly-Fish-Dev** [comment]
Tested it with [libfaketime](https://github.com/wolfcw/libfaketime), appears to have been fixed?
```
faketime "101059-11-18" ./tty-clock
```

<img width="447" height="264" alt="Image" src="https://github.com/user-attachments/assets/4dc17328-1e6c-4dee-9967-c8b95558a279" />

**Comment by Citrullin** [comment]
Please tell me that's AI slop.

**Comment by EverStarck** [comment]
valid concern btw

**Comment by lowlune** [comment]
> Please tell me that's AI slop.

you are AI slop

**Comment by sumedhahire** [comment]
This is a critical bug we have a very short amount of time left

**Comment by NecA970** [comment]
when this fails there can easily be 800 trillion dollars in damages (currently 5$ adjusted for inflation)

**Comment by transdeveloper** [comment]
<img width="336" height="499" alt="Image" src="https://github.com/user-attachments/assets/f8456ffb-29a6-41b6-a669-7fcaacacc5ab" />
billions must patch future bugs!!

**Comment by AusaafAhmad** [comment]
fair

**Comment by exilonium** [comment]
The only reason we're not using it in prod cause this will break out system in the future 

To mitigate this issue 
We're making an api request to claude to get the current time.
It works like a charm 🫡

**Comment by yarrseni** [comment]
> The only reason we're not using it in prod cause this will break out system in the future
> 
> To mitigate this issue We're making an api request to claude to get the current time. It works like a charm 🫡

You shouldn't rely on only one provider, it's quite unsafe. The requests should go to all major LLM providers and fed into a locally deployed triage agentic system of open weights LLMs (3+, but tailor to your SLAs).

**Comment by exilonium** [comment]
> You shouldn't rely on only one provider, it's quite unsafe. The requests should go to all major LLM providers and fed into a locally deployed triage agentic system of open weights LLMs (3+, but tailor to your SLAs).


That's a quite nice system that we tried to implement but claude seems to be most accurate 
We've created a weighted chain of the delay in request and response so the
Time comes out to be with delay of +-2 milliseconds that's quite accurate 

Openai gpt 4.1 has a variable response rate so we get the time with +-8ms accuracy 


Tho i think implementing the traige system will be most helpful 


Either way we're using tty clock as a fail safe in case the connection drops or we're out of ai quotas 

(It's currently costing us ~2$ per minute)

Tho i really love the engineers we're working with cause they found the solution this problem with only a single cursor prompt 🥰
(Saved us a dollar)

**Comment by Speculate7348** [comment]
Is this fixed yet??

**Comment by endreszabo** [comment]
> Is this fixed yet??

looping in for visibility. this is blocking our go-live, please fix asap and report

**Comment by trancikk** [comment]
even after 2 billions years something will crash with seg fault
eternal classic

**Comment by Minecraft633** [comment]
Well uhm, humanity has a 95% probability of being extinct in 8,000,000 years so fixing this bug will make no sense, unless aliens manage to discover this

**Comment by trancikk** [comment]
> Well uhm, humanity has a 95% probability of being extinct in 8,000,000 years so fixing this bug will make no sense, unless aliens manage to discover this

personally i would not want them to think we are just a bunch of incompetent software engineers so its better to fix this bug

**Comment by Atomosk** [comment]
~36000000-day vulnerability and the clock is ticking.
